service/kafka-plugin/src/main/java/com/solace/maas/ep/event/management/agent/plugin/kafka/route/handler/KafkaClusterConfigurationDataPublisherRouteBuilder.java

```diff
@@ -13,8 +14,9 @@
 public class KafkaClusterConfigurationDataPublisherRouteBuilder extends DataPublisherRouteBuilder {
     @Autowired
     public KafkaClusterConfigurationDataPublisherRouteBuilder(KafkaClusterConfigurationProcessor processor,
-                                                              RouteManager routeManager, MDCProcessor mdcProcessor) {
+                                                              RouteManager routeManager, MDCProcessor mdcProcessor,
+                                                              RouteCompleteProcessor routeCompleteProcessor) {
         super(processor, KafkaRouteId.KAFKA_CLUSTER_CONFIGURATION.label, KafkaRouteType.KAFKA_CUSTER_CONFIGURATION.label,
-                routeManager, mdcProcessor);
+                routeManager, mdcProcessor, routeCompleteProcessor);
```

Comment:
Doing this to all of these endpoints isn't really necessary. If you create a single RouteBuilder that handles all "Scan Status" messages you can avoid injecting your routeCompleteProcessor into every single plugin.

Take a look at the "DataCollectionFileWriteRouteBuilder" class.